backend/core/models/user.py

```diff
@@ -244,6 +245,17 @@ def ensure_email_domain_exists(self):
         domain = f"@{domain_part}".lower()
         EmailDomain.objects.get_or_create(domain=domain)
 
+    def clean(self):
+        value = self.email
+
+        similar = User.objects.filter(email__iexact=value)
+        n_similar = similar.count()
+
+        if n_similar > 1:
+            raise ValidationError(_("A user with this email address already exists."))
+        elif n_similar == 1 and similar[0].pk != self.pk:
+            raise ValidationError(_("A user with this email address already exists."))
```

Comment:
`ValidationError` accepts a dict to associate the error with a specific field in the form:
```suggestion
            raise ValidationError(
                {"email": _("A user with this email address already exists.")}
            )
```

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/812/commits/06d499b7747f8b3c93db04346e98525f6ca74f21.